### PR TITLE
Fix double submission using button[type=submit]

### LINKED ISSUE #6233: Prevent amp-form double submissions
I have a POST amp-form to add stuff to the cart, similar to:
````
<form class="cart" method="post" 
action-xhr="//local.web/product-123-page/amp/?action=add-to-cart" 
target="_blank">
	<input type="number" name="quantity" value="1" title="Qty">
	<input type="hidden" name="product_id" value="123"
        <button type="submit" class="button">Add to cart</button>
</form>
````
(markup simplified for clarity)

This works very fine, but if one clicks twice rapidly on the submit button, the following happens:

1 - The XHR request is started, adding one product to the cart (and eventually causing the display of a success message)
2 - While the XHR request is happening, the second click is handled, and the form gets submitted **a second time**. 
In addition, due to the target="_blank" attribute, a new tab is opened by the browser to complete the request. This tabs display the original page, without a success/failure message (as there was no XHR request done there).

Is there a way to suppress clicks handling until first XHR request has terminated, whether success or failure?

Thanks

### PR DESCRIPTION
**Pull request by mkhatib**
Fixes #6233

### CODE REVIEW

File: extensions/amp-form/0.1/amp-form.js [hunk 1]
```diff
@@ -118,7 +118,8 @@ export class AmpForm {
     }
     this.form_.classList.add('-amp-form');
 
-    const submitButtons = this.form_.querySelectorAll('input[type=submit]');
+    const submitButtons = this.form_.querySelectorAll(
+        'input[type="submit"],button[type="submit"]');
```
Comment (jridgewell) [inline extensions/amp-form/0.1/amp-form.js]: Just `[type="submit"]`?

File: extensions/amp-form/0.1/test/test-amp-form.js [hunk 2]
```diff
@@ -291,7 +298,7 @@ describe('amp-form', () => {
   });
 
   it('should block multiple submissions and disable buttons', () => {
```
Comment (muxin) [inline extensions/amp-form/0.1/test/test-amp-form.js]: Also add test to `should manage form state classes (submitting, error)`?

File: extensions/amp-form/0.1/amp-form.js [hunk 3]
```diff
@@ -189,6 +191,7 @@ export class AmpForm {
     if (this.shouldValidate_ && !isValid) {
       if (opt_event) {
         opt_event.stopImmediatePropagation();
+        opt_event.preventDefault();
```
Comment (muxin) [inline extensions/amp-form/0.1/amp-form.js]: Add `preventDefault` and `stopImmediatePropagation` check in the validity test?

File: extensions/amp-form/0.1/test/test-amp-form.js (hunk 2 above)
Comment (mkhatib) [inline extensions/amp-form/0.1/test/test-amp-form.js]: To test for what? I don't think changes there are needed

File: extensions/amp-form/0.1/amp-form.js (hunk 3 above)
Comment (mkhatib) [inline extensions/amp-form/0.1/amp-form.js]: I've added some more expects statements for this path.

File: extensions/amp-form/0.1/test/test-amp-form.js (hunk 2 above)
Comment (mkhatib) [inline extensions/amp-form/0.1/test/test-amp-form.js]: Added an expect statement there

**Comment by mkhatib** [comment]
PTAL 👀

File: extensions/amp-form/0.1/test/test-amp-form.js (hunk 2 above)
Comment (muxin) [inline extensions/amp-form/0.1/test/test-amp-form.js]: Oh, nevermind

**Pull request merged**